snuba/datasets/replays_processor.py

```diff
@@ -0,0 +1,170 @@
+            )
+        except EventTooOld:
+            return None
+
+        return event, retention_days
+
+    def _process_base_event_values(
+        self, processed: MutableMapping[str, Any], event_dict: EventDict
+    ) -> MutableMapping[str, Any]:
+
+        processed["replay_id"] = str(uuid.UUID(event_dict["replay_id"]))
+        processed["project_id"] = event_dict["project_id"]
+
+        processed["sequence_id"] = event_dict["sequence_id"]
+        processed["trace_ids"] = event_dict["trace_ids"]
+
+        processed["timestamp"] = self.__extract_timestamp(
+            event_dict["data"]["timestamp"],
+        )
+
+        processed["platform"] = _unicodify(event_dict["platform"])
+        return processed
+
+    def _process_tags(
+        self,
+        processed: MutableMapping[str, Any],
+        event_dict: EventDict,
+    ) -> None:
+
+        tags: Mapping[str, Any] = _as_dict_safe(event_dict["data"].get("tags", None))
+        processed["tags.key"], processed["tags.value"] = extract_extra_tags(tags)
+        promoted_tags = {col: tags[col] for col in self.PROMOTED_TAGS if col in tags}
+        processed["release"] = promoted_tags.get(
+            "sentry:release",
+            event_dict.get("release"),
+        )
+        processed["environment"] = promoted_tags.get("environment")
+        processed["user"] = promoted_tags.get("sentry:user", "")
+        processed["dist"] = _unicodify(
+            promoted_tags.get("sentry:dist", event_dict["data"].get("dist")),
+        )
+
+    def _process_user(
+        self,
+        processed: MutableMapping[str, Any],
+        event_dict: EventDict,
+    ) -> None:
+
+        user_dict = (
+            event_dict["data"].get(
+                "user", event_dict["data"].get("sentry.interfaces.User", None)
+            )
+            or {}
+        )
+
+        user_data: MutableMapping[str, Any] = {}
+        extract_user(user_data, user_dict)
+        processed["user_name"] = user_data["username"]
+        processed["user_id"] = user_data["user_id"]
+        processed["user_email"] = user_data["email"]
+        ip_address = _ensure_valid_ip(user_data["ip_address"])
+        if ip_address:
+            if ip_address.version == 4:
+                processed["ip_address_v4"] = str(ip_address)
+            elif ip_address.version == 6:
+                processed["ip_address_v6"] = str(ip_address)
+
+    def _process_sdk_data(
+        self,
+        processed: MutableMapping[str, Any],
+        event_dict: EventDict,
+    ) -> None:
+        sdk = event_dict["data"].get("sdk", None) or {}
+        processed["sdk_name"] = _unicodify(sdk.get("name") or "")
+        processed["sdk_version"] = _unicodify(sdk.get("version") or "")
+
+        if processed["sdk_name"] == "":
+            metrics.increment("missing_sdk_name")
+        if processed["sdk_version"] == "":
+            metrics.increment("missing_sdk_version")
+
+    def process_message(
```

Comment:
For now, can you put all the logic in this function into a try/except that logs a warning/sets a metric in Datadog? Currently if a bad message or something causes an exception it will crash the consumer.

Usually during the testing/experimentation phase we log errors on bad messages instead of crashing outright, and then once the data model etc. have been solidified we remove the error handling.

Reply:
Why not just applying the Dead letter queue? Now we have one. 
Just set up a dead letter queue policy when you define the storage (which is something you have to do anyway in order to consume)
https://github.com/getsentry/snuba/blob/master/snuba/datasets/storages/metrics.py#L96

Reply:
added a TODO for setting up a deadletter queue, need to provision the topic, etc.